modules/calib3d/src/usac.hpp

```diff
@@ -0,0 +1,744 @@
+    virtual Score getScore (const Mat& model, bool get_inliers, std::vector<int>& inliers) const = 0;
+    virtual Score getScore (const Mat& model) const = 0;
+
+    // make sure than estimator is updated with the latest model.
+    // make isInlier() also virtual for MAGSAC, because threshold is unknown.
+    // for other quality classes inlier is point which error is less than threshold.
+    virtual bool isInlier (int point_idx) const = 0;
+
+    // set model to estimator
+    virtual void setModel (const Mat& model) const = 0;
+
+    // get inliers with initialized threshold
+    virtual int getInliers (const Mat& model, std::vector<int>& inliers) const = 0;
+
+    // get inliers for given threshold
+    virtual int getInliers (const Mat& model, std::vector<int>& inliers, double thr) const = 0;
+
+    /*
+     * Return:
+     * @inliers: array of inliers of size number of inliers.
+     * @errors: array of points size, value under index i of array corresponds to error of point i if i is inliers,
+     *          otherwise, value not defined.
+     * @inliers_mask: true if point i is inlier, false - otherwise.
+     */
+    virtual int getInliers (const Mat& model, std::vector<int>& inliers, std::vector<bool>& inliers_mask, std::vector<double>& errors) const = 0;
+};
+
+class RansacQuality : public Quality {
+public:
+    static Ptr<RansacQuality> create(int points_size_, double threshold_, const Ptr<Error> &error_);
+};
+
+class MsacQuality : public Quality {
+public:
+    static Ptr<MsacQuality> create(int points_size_, double threshold_, const Ptr<Error> &error_);
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////// MODEL VERIFIER ////////////////////////////////////
+class ModelVerifier : public Algorithm {
+public:
+    virtual ~ModelVerifier() override = default;
+
+    /*
+     * Returns true if model is good, false - otherwise.
+     */
+    inline virtual bool isModelGood(const Mat &model) {
+        return true;
+    }
+
+    /*
+     * Return true if score was computed during evaluation.
+     */
+    inline virtual bool getScore(Score &score) const {
+        return false;
+    }
+
+    /*
+     * Reset ModelVerification to initial state.
+     * Assume parameters of ModelVerification (e.g., points size) are not changed.
+     * SPRT as sequential decision making requires to be reset.
+     */
+    virtual void reset () {
+
+    }
+
+//    virtual bool * getComputedErrorsFlags () const = 0;
+//    virtual double * getComputedErrors () const = 0;
+};
+
+/*
+* Chum, Ondrej, and Jiri Matas. "Randomized RANSAC with Td, d test."
+* Proc. British Machine Vision Conference. Vol. 2. 2002.
+*/
+class Tdd : public ModelVerifier {
+public:
+    static Ptr<Tdd> create(RNG &rng, int points_size_, const Ptr<Quality> &quality_,
+            int d_points_to_test_ = 1);
+};
+
+class SPRT_history {
```

Comment:
for stateless data structures why not use struct?

Reply:
fixed